src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IWithdrawalHistoryProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+using Stratis.FederatedPeg.Features.FederationGateway.Models;
+
+namespace Stratis.FederatedPeg.Features.FederationGateway.Interfaces
+{
+    public interface IWithdrawalHistoryProvider
```

Comment:
No need for this interface. The right place for the implementation is in the FederationWalletManager where you already have a getHistory method.